firebase-sessions/src/main/kotlin/com/google/firebase/sessions/ProcessUtils.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import android.app.ActivityManager
+import android.app.ActivityManager.RunningAppProcessInfo
+import android.app.Application
+import android.content.Context
+import android.os.Build
+import android.os.Process
+
+/** Utils related to processes. */
+internal object ProcessUtils {
+  /** Returns whether the current process should generate a new session or not. */
+  fun shouldProcessGenerateNewSession(context: Context): Boolean =
```

Comment:
Should this function move out of this file? I feel that the decision maker for this should be in something like `SessionInitiator` rather than `ProcessUtils`.

Reply:
Disagree. This needs to be checked outside of `SessionInitiator`.

Reply:
I'm fine with not doing it in SessionInitiator. The challenge is to add this function in the ProcessUtils. Ideally ProcessUtils should serve information about the process - but the decision of should I start a session or not should be made of this utility class (ideally in the place where we determine if we should start a session).

One of the choice I proposed is to do it somewhere [here](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionInitiator.kt#L58-L65) in `SessionInitiator` class. If we have a better place, totally fine.

Reply:
Yeah, that class comes in to play after we've decided that the process is going to generate sessions. It won't be involved at all if the process should not generate sessions.